src/tss2-esys/esys_iutil.c

```diff
@@ -552,6 +552,85 @@ iesys_gen_caller_nonces(ESYS_CONTEXT * esys_context)
     }
     return TSS2_RC_SUCCESS;
 }
+
+/** Update session attributes.
+ *
+ * In case where command does not support param encryption/decryption
+ * store the original session attributes and update them accordingly.
+ * Return true is command support param encryption.
+ *
+ * @retval TRUE if command support param encryption
+ * @retval FLASE if command does not support param encryption
+ */
+static bool
+iesys_update_session_flags(ESYS_CONTEXT * esys_context,
+                           IESYS_SESSION *rsrc_session)
+{
+    TSS2_RC r = TSS2_RC_SUCCESS;
+    size_t param_size;
+    const uint8_t *param_buffer;
+    bool encrypt = true;
```

Comment:
no need for this AFAICT

Reply:
We do need this. It's no good if we unset the flags, but still encrypt the parameter.